packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -270,9 +271,18 @@ export const createWebpackUtils = (
     },
 
     js: options => {
+      let reactMajorVersion = 0
+
+      try {
+        const react = require(`react`)
+        reactMajorVersion = semver.major(react.version)
+      } catch (e) {
+        // no-op.
+      }
       return {
         options: {
           stage,
+          runtime: reactMajorVersion >= 17 ? `automatic` : `classic`,
```

Comment:
Let's be more specific and call it reactRuntime instead of just runtime :) Runtime can mean many things.
```suggestion
          reactRuntime: reactMajorVersion >= 17 ? `automatic` : `classic`,
```